pkg/mempool/custom_proposal_handler_test.go

```diff
@@ -0,0 +1,393 @@
+		},
+	}
+
+	// create context
+	s.ctx = sdk.Context{}.WithConsensusParams(&cmtproto.ConsensusParams{})
+	return s
+}
+
+func TestABCIUtilsTestSuite(t *testing.T) {
+	suite.Run(t, NewABCIUtilsTestSuite(t))
+}
+
+func (s *ABCIUtilsTestSuite) TestCustomProposalHandler_NoopMempool() {
+	cdc := codec.NewProtoCodec(codectypes.NewInterfaceRegistry())
+	baseapptestutil.RegisterInterfaces(cdc.InterfaceRegistry())
+	txConfig := authtx.NewTxConfig(cdc, authtx.DefaultSignModes)
+	var (
+		secret1 = []byte("secret1")
+		secret2 = []byte("secret2")
+	)
+	type testTx struct {
+		tx       sdk.Tx
+		priority int64
+		bz       []byte
+	}
+	testTxs := []testTx{
+		{tx: buildMsg(s.T(), txConfig, []byte(`0`), [][]byte{secret1}, []uint64{1}), priority: 10},
+		{tx: buildMsg(s.T(), txConfig, []byte(`12345678910`), [][]byte{secret1}, []uint64{2}), priority: 10},
+		{tx: buildMsg(s.T(), txConfig, []byte(`22`), [][]byte{secret1}, []uint64{3}), priority: 10},
+		{tx: buildMsg(s.T(), txConfig, []byte(`32`), [][]byte{secret2}, []uint64{1}), priority: 8},
+	}
+	ctrl := gomock.NewController(s.T())
+	app := mock.NewMockProposalTxVerifier(ctrl)
+	mp := mempool.NoOpMempool{}
+	customProposalHandler := zetamempool.NewCustomProposalHandler(mp, app)
+	prepareProposalHandler := customProposalHandler.PrepareProposalHandler()
+	processProposalHandler := customProposalHandler.ProcessProposalHandler()
+
+	req := abci.RequestPrepareProposal{
+		MaxTxBytes: 111 + 112,
+	}
+	for _, v := range testTxs {
+		app.EXPECT().PrepareProposalVerifyTx(v.tx).Return(v.bz, nil).AnyTimes()
+		s.NoError(mp.Insert(s.ctx.WithPriority(v.priority), v.tx))
+		req.Txs = append(req.Txs, v.bz)
+	}
+
+	// since it is noop mempool, all txs returned in fifo order
+	resp := prepareProposalHandler(s.ctx, req)
+	for i, tx := range resp.Txs {
+		require.Equal(s.T(), tx, testTxs[i].bz)
+	}
+
+	// verify that proposal will be processed
+	for i, v := range req.Txs {
+		app.EXPECT().ProcessProposalVerifyTx(v).Return(testTxs[i].tx, nil).AnyTimes()
+	}
+	resp2 := processProposalHandler(s.ctx, abci.RequestProcessProposal{
+		Txs: resp.Txs,
+	})
+	s.Require().Equal(resp2.Status, abci.ResponseProcessProposal_ACCEPT)
+}
+
+func (s *ABCIUtilsTestSuite) TestCustomProposalHandler_PriorityNonceMempoolTxSelection() {
+	cdc := codec.NewProtoCodec(codectypes.NewInterfaceRegistry())
+	baseapptestutil.RegisterInterfaces(cdc.InterfaceRegistry())
+	txConfig := authtx.NewTxConfig(cdc, authtx.DefaultSignModes)
+	var (
+		secret1 = []byte("secret1")
+		secret2 = []byte("secret2")
+		secret3 = []byte("secret3")
+		secret4 = []byte("secret4")
+		secret5 = []byte("secret5")
+		secret6 = []byte("secret6")
+	)
+
+	type testTx struct {
+		tx       sdk.Tx
+		priority int64
+		bz       []byte
+		size     int
+	}
+
+	testTxs := []testTx{
+		// test 1
+		{tx: buildMsg(s.T(), txConfig, []byte(`0`), [][]byte{secret1}, []uint64{1}), priority: 10},
+		{tx: buildMsg(s.T(), txConfig, []byte(`12345678910`), [][]byte{secret1}, []uint64{2}), priority: 10},
```

Comment:
NIT : maybe we can use a constant here and in the other cases , might make it easier to read 
`MoreThanMaxBytes = []byte(`12345678910`)`

Reply:
Hmmm , is there a reason we need the different sizes  ?
But based on the other comments, I think it would make sense to keep the test as long as it is able to test the functionality